docs/source/cc_launcher.md

```diff
@@ -0,0 +1,301 @@
+```
+
+When `build` is invoked, `CHAINCODE_SOURCE_DIR` contains the chaincode source
+and `CHAINCODE_METADATA_DIR` contains the `metadata.json` file from the
+chaincode package installed to the peer. `BUILD_OUTPUT_DIR` is the directory
+where `build` must place artifacts needed by `release` and `run`.
+
+When `build` completes with an exit code of `0`, the contents of
+`BUILD_OUTPUT_DIR` will be copied to the persistent storage maintained by the
+peer; any other exit code will be considered a failure.
+
+The following is an example of a simple `build` script for go chaincode:
+```sh
+#!/bin/bash
+
+CHAINCODE_SOURCE_DIR="$1"
+CHAINCODE_METADATA_DIR="$2"
+BUILD_OUTPUT_DIR="$3"
+
+# extract package path from metadata.json
+GO_PACKAGE_PATH="$(jq -r .path "$CHAINCODE_METADATA_DIR/metadata.json")"
+if [ -f "$CHAINCODE_SOURCE_DIR/src/go.mod" ]; then
+    cd "$CHAINCODE_SOURCE_DIR/src"
+    go build -v -mod=readonly -o "$BUILD_OUTPUT_DIR/chaincode" "$GO_PACKAGE_PATH"
+else
+    GO111MODULE=off go build -v  -o "$BUILD_OUTPUT_DIR/chaincode" "$GO_PACKAGE_PATH"
+fi
+
+# save statedb index metadata to provide at release
+if [ -d "$CHAINCODE_SOURCE_DIR/META-INF" ]; then
+    cp -a "$CHAINCODE_SOURCE_DIR/META-INF" "$BUILD_OUTPUT_DIR/"
+fi
+```
+
+#### `bin/release`
+
+The `bin/release` script is responsible for providing metadata chaincode to
+the peer. The peer invokes `release` with two arguments:
+
+```sh
+bin/release BUILD_OUTPUT_DIR RELEASE_OUTPUT_DIR
+```
+
+When `release` is invoked, `BUILD_OUTPUT_DIR` contains the artifacts populated
+by the `build` program. `RELEASE_OUTPUT_DIR` is the directory where `release`
+must place artifacts to be consumed by the peer.
+
+When `release` completes, the peer will consume two types of metadata from
+`RELEASE_OUTPUT_DIR`:
+
+- state database index definitions for CouchDB
+- external chaincode server connection information (`chaincode/server/connection.json`)
+
+If CouchDB index definitions required for the chaincode, `release` is
+responsible for placing the indexes into the `statedb/couchdb/indexes`
+directory under `RELEASE_OUTPUT_DIR`. The indexes must have a `.json`
+extension.  See the [CouchDB indexes][couchdb-indexes] documentation for
+details.
+
+In cases where a chaincode server implementation is used, `release` is
+responsible for populating `chaincode/server/connection.json` with the address
+of the chaincode server and any TLS assets required to communicate with the
+chaincode. When server connection information is provided to the peer, `run`
+will not be called. See the [Chaincode Server][chaincode-server]
+documentation for details.
+
+The following is an example of a simple `release` script for go chaincode:
+```sh
+#!/bin/bash
+
+BUILD_OUTPUT_DIR="$1"
+RELEASE_OUTPUT_DIR="$2"
+
+# copy indexes from META-INF/* to the output directory
+if [ -d "$BUILD_OUTPUT_DIR/META-INF" ] ; then
+   cp -a "$BUILD_OUTPUT_DIR/META-INF/"* "$RELEASE_OUTPUT_DIR/"
+fi
+```
+
+#### `bin/run`
+
```

Comment:
Maybe note that 'run' should treat its parameter directories as read-only.

Reply:
Done.